File: BaekJoon/ITBigTech/Rank1205/Rank.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func main() {
	reader := bufio.NewReader(os.Stdin)
	userInput, _ := reader.ReadString('\n')
	userInput = strings.TrimSpace(userInput)
	input := strings.Split(userInput, " ")
	
	N, _ := strconv.Atoi(input[0])
	newScore, _ := strconv.Atoi(input[1])
	P, _ := strconv.Atoi(input[2])
    
    rank := 1
    
    var rankList []int
	listInput, _ := reader.ReadString('\n')
	listInput = strings.TrimSpace(listInput)
	rankInput := strings.Split(listInput, " ")
    
    if N > 0 {
        for i := 0; i < N; i++ {
            rank, _ := strconv.Atoi(rankInput[i])
            rankList = append(rankList, rank) 
        }
    }

    for i := 0; i < N; i++ {
        if newScore > rankList[i] {
            break
        } else if newScore == rankList[i] {
            break
        } else {
            rank++
        }
    }

    if N == P && N > 0 && newScore <= rankList[N-1] {
        fmt.Println(-1)
    } else {
        fmt.Println(rank)
    }
}

File: BaekJoon/ITBigTech/Rank1205/Rank.py
N, new, P = map(int, input().split())

if N > 0:
    rank_list = list(map(int, input().split()))
else:
    rank_list = []

rank = 1

# 리스트로 들어가봅시다
for i in range(N):
    if new > rank_list[i]:
        break
    elif new == rank_list[i]:
        break
    else:
        rank += 1

# 꽉 차있을 때 마지막과 같을 경우 못넣음
if N == P and N > 0 and new <= rank_list[-1]:
    print(-1)
else:
    print(rank)
